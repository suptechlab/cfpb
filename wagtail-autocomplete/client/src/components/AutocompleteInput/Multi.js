import React, { PureComponent } from "react";

import { nc } from ".";
import { RemoveIcon } from "./Icons";
import Suggestions from "./Suggestions";

class Multi extends PureComponent {
  constructor(...args) {
    super(...args);

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(suggestion) {
    const { onClick, selections } = this.props;
    onClick(selections.concat(suggestion));
  }

  handleRemove(page) {
    const { onClick, selections } = this.props;
    onClick(selections.filter(({ pk }) => pk !== page.pk));
  }

  render() {
    const {
      selections,
      onChange,
      onCreate,
      canCreate,
      input,
      labelId
    } = this.props;

    const suggestions = this.props.suggestions.filter(suggestion => {
      if (!selections) {
        return true;
      }
      return selections.every(({ pk }) => pk !== suggestion.pk);
    });

    return (
      <span className={nc("layout")}>
        <span className={nc("layout__item")}>
          <Suggestions
            labelId={labelId}
            suggestions={suggestions}
            onClick={this.handleClick}
            onChange={onChange}
            onCreate={onCreate}
            canCreate={canCreate}
            input={input}
            inputElm={this.inputElm}
          />
        </span>

        <span className={nc("layout__item", "layout__item--padded")}>
          {selections.length === 0 && <span>Nothing selected.</span>}
          {selections.map(selection => (
            <div key={selection.pk} className={nc("selection")}>
              <span className={nc("selection__label")}>{selection.title}</span>

              <button
                type="button"
                className={nc("selection__button")}
                onClick={this.handleRemove.bind(this, selection)}
              >
                <RemoveIcon className={nc("selection__icon")} />

                <span className={nc("sr-only")}>Remove</span>
              </button>
            </div>
          ))}
        </span>
      </span>
    );
  }
}

Multi.defaultProps = {
  selections: []
};

export default Multi;
