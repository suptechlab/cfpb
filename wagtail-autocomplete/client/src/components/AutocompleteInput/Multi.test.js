import React from 'react';
import { shallow } from 'enzyme';

import Multi from './Multi';


const mockProps = {
  suggestions: [
    { id: 1, title: 'Alice' },
    { id: 2, title: 'Tekisha' },
  ],
  selected: [],
  onChange: jest.fn(),
  onCreate: jest.fn(),
  onClick: jest.fn(),
  input: { value: '' },
  canCreate: true,
};


describe('Multi', () => {
  it('exists', () => {
    expect(Multi).toBeDefined();
  });

  it('mounts', () => {
    const multi = shallow(
      <Multi {...mockProps} />
    );
    expect(multi).toMatchSnapshot();
  });
});
