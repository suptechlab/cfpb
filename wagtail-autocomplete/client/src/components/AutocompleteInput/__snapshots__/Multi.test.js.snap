// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multi mounts 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <span
    className="c-wagtailautocomplete__layout"
>
    <span
        className="c-wagtailautocomplete__layout__item"
    >
        <Suggestions
            canCreate={true}
            input={
                Object {
                    "value": "",
                  }
            }
            inputElm={undefined}
            labelId={undefined}
            onChange={[Function]}
            onClick={[Function]}
            onCreate={[Function]}
            suggestions={
                Array [
                    Object {
                      "id": 1,
                      "title": "Alice",
                    },
                    Object {
                      "id": 2,
                      "title": "Tekisha",
                    },
                  ]
            }
        />
    </span>
    <span
        className="c-wagtailautocomplete__layout__item"
    >
        <span>
            Nothing selected.
        </span>
    </span>
</span>,
  "nodes": Array [
    <span
      className="c-wagtailautocomplete__layout"
>
      <span
            className="c-wagtailautocomplete__layout__item"
      >
            <Suggestions
                  canCreate={true}
                  input={
                        Object {
                              "value": "",
                            }
                  }
                  inputElm={undefined}
                  labelId={undefined}
                  onChange={[Function]}
                  onClick={[Function]}
                  onCreate={[Function]}
                  suggestions={
                        Array [
                              Object {
                                "id": 1,
                                "title": "Alice",
                              },
                              Object {
                                "id": 2,
                                "title": "Tekisha",
                              },
                            ]
                  }
            />
      </span>
      <span
            className="c-wagtailautocomplete__layout__item"
      >
            <span>
                  Nothing selected.
            </span>
      </span>
</span>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Multi
        canCreate={true}
        input={
                Object {
                        "value": "",
                      }
        }
        onChange={[Function]}
        onClick={[Function]}
        onCreate={[Function]}
        selected={Array []}
        selections={Array []}
        suggestions={
                Array [
                        Object {
                          "id": 1,
                          "title": "Alice",
                        },
                        Object {
                          "id": 2,
                          "title": "Tekisha",
                        },
                      ]
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Multi {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleClick": [Function],
        "props": Object {
          "canCreate": true,
          "input": Object {
            "value": "",
          },
          "onChange": [Function],
          "onClick": [Function],
          "onCreate": [Function],
          "selected": Array [],
          "selections": Array [],
          "suggestions": Array [
            Object {
              "id": 1,
              "title": "Alice",
            },
            Object {
              "id": 2,
              "title": "Tekisha",
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <span
          className="c-wagtailautocomplete__layout"
>
          <span
                    className="c-wagtailautocomplete__layout__item"
          >
                    <Suggestions
                              canCreate={true}
                              input={
                                        Object {
                                                  "value": "",
                                                }
                              }
                              inputElm={undefined}
                              labelId={undefined}
                              onChange={[Function]}
                              onClick={[Function]}
                              onCreate={[Function]}
                              suggestions={
                                        Array [
                                                  Object {
                                                    "id": 1,
                                                    "title": "Alice",
                                                  },
                                                  Object {
                                                    "id": 2,
                                                    "title": "Tekisha",
                                                  },
                                                ]
                              }
                    />
          </span>
          <span
                    className="c-wagtailautocomplete__layout__item"
          >
                    <span>
                              Nothing selected.
                    </span>
          </span>
</span>,
        "_debugID": 2,
        "_renderedOutput": <span
          className="c-wagtailautocomplete__layout"
>
          <span
                    className="c-wagtailautocomplete__layout__item"
          >
                    <Suggestions
                              canCreate={true}
                              input={
                                        Object {
                                                  "value": "",
                                                }
                              }
                              inputElm={undefined}
                              labelId={undefined}
                              onChange={[Function]}
                              onClick={[Function]}
                              onCreate={[Function]}
                              suggestions={
                                        Array [
                                                  Object {
                                                    "id": 1,
                                                    "title": "Alice",
                                                  },
                                                  Object {
                                                    "id": 2,
                                                    "title": "Tekisha",
                                                  },
                                                ]
                              }
                    />
          </span>
          <span
                    className="c-wagtailautocomplete__layout__item"
          >
                    <span>
                              Nothing selected.
                    </span>
          </span>
</span>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Multi
    canCreate={true}
    input={
        Object {
            "value": "",
          }
    }
    onChange={[Function]}
    onClick={[Function]}
    onCreate={[Function]}
    selected={Array []}
    selections={Array []}
    suggestions={
        Array [
            Object {
              "id": 1,
              "title": "Alice",
            },
            Object {
              "id": 2,
              "title": "Tekisha",
            },
          ]
    }
/>,
}
`;
